metadata-ingestion/src/datahub/cli/quickstart_versioning.py

```diff
@@ -0,0 +1,145 @@
+import json
+import os
+import re
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple
+
+import click
+import requests
+import yaml
+from pydantic import BaseModel, PrivateAttr
+
+DEFAULT_LOCAL_CONFIG_PATH = "~/.datahub/quickstart/quickstart_version_mapping.yaml"
+DEFAULT_REMOTE_CONFIG_PATH = "https://raw.githubusercontent.com/datahub-project/datahub/quickstart-stability/docker/quickstart/quickstart_version_mapping.yaml"
+
+
+class QuickstartExecutionPlan(BaseModel):
+    composefile_git_ref: str
+    docker_tag: str
+
+
+def is_it_a_version(version: str) -> bool:
+    """
+    Checks if a string is a valid version.
+    :param version: The string to check.
+    :return: True if the string is a valid version, False otherwise.
+    """
+    return re.match(r"v?\d+\.\d+(\.\d+)?", version) is not None
+
+
+def parse_version(version: str) -> Tuple[int, int, int, int]:
+    """
+    Parses a version string into a tuple of integers.
+    :param version: The version string to parse.
+    :return: A tuple of integers representing the version.
+    """
+    version = re.sub(r"v", "", version)
+    parsed_version = tuple(map(int, version.split(".")))
```

Comment:
let's use packaging.version.parse()

the object it returns also natively supports comparison